pkg/etcd/etcd.go

```diff
@@ -633,34 +633,27 @@ func (e *ETCD) setSnapshotFunction(ctx context.Context) {
 // completion.
 func (e *ETCD) Restore(ctx context.Context) error {
 	// check the old etcd data dir
-	oldDataDir := dataDir(e.config) + "-old"
-	if s, err := os.Stat(oldDataDir); err == nil && s.IsDir() {
-		logrus.Infof("Etcd already restored from a snapshot. Restart without --snapshot-restore-path flag. Backup and delete ${datadir}/server/db on each peer etcd server and rejoin the nodes")
-		os.Exit(0)
-	} else if os.IsNotExist(err) {
-		if e.config.ClusterResetRestorePath == "" {
-			return errors.New("no etcd restore path was specified")
-		}
-		// make sure snapshot exists before restoration
-		if _, err := os.Stat(e.config.ClusterResetRestorePath); err != nil {
-			return err
-		}
-		// move the data directory to a temp path
-		if err := os.Rename(dataDir(e.config), oldDataDir); err != nil {
-			return err
-		}
-		sManager := snapshot.NewV3(nil)
-		if err := sManager.Restore(snapshot.RestoreConfig{
-			SnapshotPath:   e.config.ClusterResetRestorePath,
-			Name:           e.name,
-			OutputDataDir:  dataDir(e.config),
-			OutputWALDir:   walDir(e.config),
-			PeerURLs:       []string{e.peerURL()},
-			InitialCluster: e.name + "=" + e.peerURL(),
-		}); err != nil {
-			return err
-		}
-	} else {
+	oldDataDir := dataDir(e.config) + "-old-" + strconv.Itoa(int(time.Now().Unix()))
```

Comment:
Can we add a log line that informs the user that this directory exists and that their data from before the backup lives here? My reasoning is that we currently arent cleaning up the -old- directories, so it would at least be nice to log their existence.

Also, can you create an issue for cleaning up -old- directories over time? Doesn't need fixed right now, but dont want to lose track of it.

Reply:
sure will do